### PR TITLE
Make OptionData#addChoice accept long instead of int

### PR DESCRIPTION
**Pull request by Xirado**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing

## Pull Request Etiquette

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines][contributing].

### Changes

- [x] Internal code
- [x] Library interface (affecting end-user code) 
- [x] Documentation
- [ ] Other: \_____ 

Closes Issue: NaN

## Description

This pull request makes `OptionData#addChoice` accept a `long` instead of an `int`

### CODE REVIEW

**Comment by Andre601** [comment]
duplicate of #1815

**Comment by Xirado** [comment]
> duplicate of #1807

@Andre601 Well good thing i opened and then closed the PR you just mentioned because i selected the wrong branch

**Comment by DV8FromTheWorld** [comment]
Conflicts need to be resolved and then I can accept this

**Comment by Xirado** [comment]
@DV8FromTheWorld done

**Comment by DV8FromTheWorld** [comment]
I Don't think your merge was successful.

There are now 2 instances of `addChoice(String, long)` and one of them incorrectly points to `MAX_NAME_LENGTH` for it's Checks.check instead of `MAX_CHOICE_NAME_LENGTH`

File: src/main/java/net/dv8tion/jda/api/interactions/commands/build/OptionData.java [hunk 1]
```diff
@@ -315,38 +314,7 @@ public OptionData addChoice(@Nonnull String name, double value)
...
-     * @param  name
-     *         The name used in the client
-     * @param  value
-     *         The value received in {@link net.dv8tion.jda.api.interactions.commands.OptionMapping OptionMapping}
-     *
-     * @throws IllegalArgumentException
-     *         If any of the following checks fail
-     *         <ul>
-     *             <li>{@code name} is not null, empty and less or equal to {@value #MAX_CHOICE_NAME_LENGTH} characters long</li>
+     *             <li>{@code name} is not null, empty and less or equal to {@value #MAX_NAME_LENGTH} characters long</li>
```
Comment (DV8FromTheWorld) [inline src/main/java/net/dv8tion/jda/api/interactions/commands/build/OptionData.java:317]:
```suggestion
     *             <li>{@code name} is not null, empty and less or equal to {@value #MAX_CHOICE_NAME_LENGTH} characters long</li>
```

File: src/main/java/net/dv8tion/jda/api/interactions/commands/build/OptionData.java [hunk 2]
```diff
@@ -358,12 +326,12 @@ public OptionData addChoice(@Nonnull String name, int value)
...
     {
         Checks.notEmpty(name, "Name");
-        Checks.notLonger(name, MAX_NAME_LENGTH, "Name");
+        Checks.notLonger(name, MAX_CHOICE_NAME_LENGTH, "Name");
         Checks.check(value >= MIN_NEGATIVE_NUMBER, "Long value may not be lower than %f", MIN_NEGATIVE_NUMBER);
         Checks.check(value <= MAX_POSITIVE_NUMBER, "Long value may not be larger than %f", MAX_POSITIVE_NUMBER);
         Checks.check(choices.size() < MAX_CHOICES, "Cannot have more than 25 choices for an option!");
         if (type != OptionType.INTEGER)
-            throw new IllegalArgumentException("Cannot add long choice for OptionType." + type);
+            throw new IllegalArgumentException("Cannot add Long choice for OptionType." + type);
```
Comment (DV8FromTheWorld) [inline src/main/java/net/dv8tion/jda/api/interactions/commands/build/OptionData.java:334]:
```suggestion
            throw new IllegalArgumentException("Cannot add long choice for OptionType." + type);
```

**Pull request merged**